Tests/AdyenTests/Adyen Tests/Components/Await Component/PollingComponentTests.swift

```diff
@@ -88,7 +88,46 @@ class PollingComponentTests: XCTestCase {
         XCTAssertEqual(apiClient.counter, 3)
     }
 
-    func testNotRetryWhenRequestFails() {
+    func testRetryWhenResultFails() {
+        let apiClient = APIClientMock()
+        let retryApiClient = RetryAPIClient(apiClient: apiClient, scheduler: SimpleScheduler(maximumCount: 5))
+        let sut = PollingComponent(apiContext: Dummy.context, apiClient: retryApiClient)
+
+        let delegate = ActionComponentDelegateMock()
+        delegate.onDidFail = { _, _ in
+            XCTFail()
+        }
+
+        let onDidProvideExpectation = expectation(description: "ActionComponentDelegate.didProvide must be called.")
+        onDidProvideExpectation.assertForOverFulfill = true
+        delegate.onDidProvide = { data, component in
+            XCTAssertTrue(component === sut)
+
+            XCTAssertEqual(data.paymentData, "data")
+            XCTAssertNotNil(data.details as? AwaitActionDetails)
+
+            let details = data.details as! AwaitActionDetails
+
+            XCTAssertEqual(details.payload, "pay load")
+
+            onDidProvideExpectation.fulfill()
+        }
+
+        sut.delegate = delegate
+
+        let resultPending = MockedResult.success(PaymentStatusResponse(payload: "pay load", resultCode: .pending))
+        let resultError = MockedResult.failure(Dummy.error)
+
+        apiClient.mockedResults = [resultPending, resultError, resultPending, resultError, resultPending]
```

Comment:
can you try two error in a row?
```Swift
[resultPending, resultError, resultError, resultPending, resultError, resultPending]
```

Reply:
I have dedicated test case for this [here](https://github.com/Adyen/adyen-ios/pull/693/files/69ea72ec62fc329527c31fc806f48c7beda952e3..f303856116f64c74dbc6dfd2d3ed608de0f5424f#diff-ba5bf0c968019fc6d33a3a43a4cd478490c3feed0791e24ff9c4cf12c2140a8cR154)